Added open list state info

Geerthan opened the pull request:
Passes State Registry into each openlist via remove_min(), allowing open lists to use state info to make decisions. Prints state ID based on registry.

Geerthan committed:
Added state registry to alternation open list

Geerthan committed:
Added state info to open list remove_min function

mbrotos committed:
Squashed commit of the following:

commit 69dff971382921d308a59934a28619878fbbb790
Author: Adam Sorrenti <adamsorrenti8@gmail.com>
Date:   Thu Apr 4 12:11:29 2024 -0400

    Latex table generator init

commit 8488de9123cf84b774e335533770ebfd1e7c9b98
Author: Adam Sorrenti <adamsorrenti8@gmail.com>
Date:   Thu Apr 4 10:24:37 2024 -0400

    Update parse_plan_log.py

mbrotos committed:
Revert End of Line sequence change for readability

mbrotos commented:
https://github.com/mbrotos/downward-dynamic-explore/pull/4/commits/d647a4d9be7c8c47aa3c6bab208746fc32f20674
Merges .gitignore changes from main to help my env.

https://github.com/mbrotos/downward-dynamic-explore/pull/4/commits/fc4fc4e6a2b89eed75a4de445465d3d0f762d040
Reverts changes made to the end of line operator. This change made it impossible to see the specific changes you made in this PR. See this for more info: https://medium.com/@csmunuku/windows-and-linux-eol-sequence-configure-vs-code-and-git-37be98ef71df

Geerthan committed:
Updated to perform decision-learning in alternation open list

Geerthan committed:
Merge branch 'mbrotos:main' into open_list_stateinfo

Geerthan committed:
Merged

Geerthan committed:
Fixed merge

mbrotos committed:
Removes trailing whitespace characters

mbrotos committed:
Make epsilon_greedy, fully random

mbrotos committed:
Default probs 50/50

mbrotos committed:
Fixed indexing with double & max_hist_size bug

The last_list was a double and also used as the index value into the deque. Although the forloops tried indexing with ints.

Also, the max_hist_size check which pops front, didn't index into the list. Thus, past_g.size() is always 2.

mbrotos committed:
Fix deque type

mbrotos commented:
Commit [201e8df](https://github.com/mbrotos/downward-dynamic-explore/pull/4/commits/201e8df2c8347ade6abeb0479e75f5e18531fa46) results:

[planner_outputs-baseline-type-gbfs-2024-04-21.zip](https://drive.google.com/open?id=1LTRRsVtMxHyIlWfUL45CcTqfV01fxjmQ&usp=drive_fs)

[planner_outputs-stateinfo-type-gbfs-e-greedy-2024-04-21.zip](https://drive.google.com/open?id=1L6_qKcL7vo6krKco6KiYpmVrsqccbBFQ&usp=drive_fs)

mbrotos committed:
Merge branch 'main' into pr/4
